cocotb/triggers.py

```diff
@@ -915,9 +918,21 @@ async def with_timeout(trigger, timeout_time, timeout_unit="step"):
 
     .. versionadded:: 1.3
 
+    .. versionchanged:: 1.6
+        Support passing :term:`python:coroutine`\ s.
+
     .. deprecated:: 1.5
         Using None as the the *timeout_unit* argument is deprecated, use ``'step'`` instead.
-   """
+    """
+
+    if inspect.iscoroutine(trigger):
```

Comment:
I think this `if` should move down to immediately below `res = await First`, and not recurse.